src/components/Divider.tsx

```diff
@@ -1,50 +1,81 @@
 import classNames from 'classnames'
-import React, { useEffect } from 'react'
+import React, { useEffect, useState } from 'react'
 import { useDispatch } from 'react-redux'
 import Path from '../@types/Path'
 import { setCursorActionCreator as setCursor } from '../actions/setCursor'
-import { DIVIDER_MIN_WIDTH, DIVIDER_PLUS_PX } from '../constants'
+import rootedParentOf from '../selectors/rootedParentOf'
+import store from '../stores/app'
+import editingValueStore from '../stores/editingValue'
 import fastClick from '../util/fastClick'
+import hashPath from '../util/hashPath'
 import head from '../util/head'
 
+const DIVIDER_PLUS_PX = 20
+const DIVIDER_MIN_WIDTH = 85
+
 /** A custom horizontal rule. */
 const Divider = ({ path }: { path: Path }) => {
-  const dividerSetWidth = React.createRef<HTMLInputElement>()
+  const dividerRef = React.createRef<HTMLInputElement>()
   const dispatch = useDispatch()
+  const [width, setWidth] = useState(DIVIDER_MIN_WIDTH)
 
   /** Sets the cursor to the divider. */
   const setCursorToDivider = (e: React.MouseEvent | React.TouchEvent) => {
     e.stopPropagation()
     dispatch(setCursor({ path }))
   }
 
-  /** Get the max width of nearby for divider list child elements, add 30 px and set this width for divider. */
+  /** Get the max width of nearby elements, add DIVIDER_PLUS_PX and set this width for divider. */
   const setStyle = () => {
-    if (dividerSetWidth.current) {
-      const parentUl = dividerSetWidth.current.closest('ul')
-      const children = parentUl ? (Array.from(parentUl.childNodes) as HTMLElement[]) : []
-      const widths = children.map((child: HTMLElement) => {
-        if (child.classList.contains('child-divider')) return DIVIDER_PLUS_PX
-        const subs = child.getElementsByClassName('subthought') as HTMLCollectionOf<HTMLElement>
-        return subs.length ? subs[0].offsetWidth + DIVIDER_PLUS_PX : DIVIDER_PLUS_PX
-      })
-      const maxWidth = Math.max(...widths)
-      dividerSetWidth.current.style.width = `${maxWidth > DIVIDER_MIN_WIDTH ? maxWidth : DIVIDER_MIN_WIDTH}px`
+    if (dividerRef.current) {
+      const state = store.getState()
+      const parentPath = rootedParentOf(state, path)
+
+      const treeNode = dividerRef.current.closest('div.tree-node') as HTMLElement
+      if (!treeNode) throw new Error('Divider tree node not found')
+
+      const thoughtPath = treeNode.dataset.path
+      if (!thoughtPath || thoughtPath === '') throw new Error('Divider path not found on tree node')
+
+      let elements = document.querySelectorAll(`.tree-node[data-path^="${hashPath(parentPath)}"]:not(.thought-divider)`)
```

Comment:
First we find all tree-nodes with a path starting with the divider's parent which in a table should initially include children in both columns, but exclude any dividers so they aren't used in the width calculation.

When we do this, if elements.length is:

0: The parentPath of a root/top-level divider is empty, so our query doesn't find any elements to compare - we find top-level nodes instead to calculate width in this case `(!elements.length)`

1: Our original query gives us only one other element at this path, meaning this is likely a single divider and we're only finding its col1 counterpart, so we go up another level and use those elements `(elements === 1)`

If elements.length > 1, then we have siblings to check.

A lot of this doesn't really matter now since I'll be doing a more React-centric refactor, but let me know if I misunderstood anything.

Reply:
Ah, I see. I neglected to see the `data-path^=` attribute selector and was mistakenly thinking `data-path=`. That explains why the parent is selected.